### PR TITLE
Coalignment functionality refactor

### PR DESCRIPTION
**Pull request by wafels**
This is a refactoring of the existing mapcube co-alignment functionality.  It was inspired by comments made on the solar de-rotation of mapcubes.

The main coalignment function had a return value which changed type and meaning depending on the keywords it was passed:
```
newmc = mapcube_coalign_by_match_template(mc)
newmc, displacements = mapcube_coalign_by_match_template(mc, with_displacements=True)
displacements = mapcube_coalign_by_match_template(displacements_only=True)
```
The refactor provides the same functionality by splitting up the calculation of the shifts required for the coalignment from the manipulation of the mapcube to get a co-aligned datacube.

So now the function mapcube_coalign_by_match_template returns ONLY a co-aligned mapcube.  The NEW function calculate_match_template_shift calculates the shifts required to do the co-alignment.  The test suite and the guide docs have been updated to reflect this change.

Other changes use of the quantity input decorator and PEP8 style changes.

### CODE REVIEW

**Comment by Cadair** [comment]
@wafels Did this code get into 0.5?

You need a changelog.

File: sunpy/image/coalignment.py [hunk 1]
```diff
@@ -381,34 +520,14 @@ def mapcube_coalign_by_match_template(mc, template=None, layer_index=0,
...
-        in the same format as that returned using the
-        return_displacements_only option.  Can be used when you want to appl
-        the same displacements to multiple mapcubes.
-
-    with_displacements : bool
-        If True, return the x and y displacements applied to the input data in
-        the same format as that returned using the return_displacements_only
-        option, along with the coaligned mapcube.  The format of the return is
-        (mapcube, displacements).
-
```
Comment (Cadair) [inline sunpy/image/coalignment.py]: You need to doc `shift` here.

**Comment by Cadair** [comment]
@wafels It appears you have a conflict, you should merge master in.

**Comment by Cadair** [comment]
@wafels is it worth putting in an example of calculating the shifts explicitly?

**Comment by wafels** [comment]
Changing typos in the docs for mapcubes should NOT cause http errors in the GOES test suite.  What's up with that?

**Comment by Cadair** [comment]
:+1:

**Comment by ayshih** [comment]
Please clean up `test_coalignment.py` to use pytest fixtures instead of global statements

File: astropy_helpers [hunk 2]
```diff
@@ -1 +1 @@
-Subproject commit 2f1a7837ff42ce090656f13b9f633f841381eb76
+Subproject commit fd7cca2b3340f8ce968f60da89b8c36b93575918
```
Comment (Cadair) [inline astropy_helpers]: this needs removing.

**Comment by Cadair** [comment]
@ehsteve @ayshih @cadair Please review.

**Comment by ehsteve** [comment]
:+1:

**Comment by wafels** [comment]
Hoorah!

**Pull request merged**